config/detekt/detekt.yml

```diff
@@ -9,8 +9,8 @@ build:
 
 config:
   validation: true
-  # when writing own rules with new properties, exclude the property path e.g.: "my_rule_set,.*>.*>[my_property]"
-  excludes: ""
+  # when writing own rules with new properties, exclude the property path e.g.: 'my_rule_set,.*>.*>[my_property]'
+  excludes: ''
```

Comment:
So why dos stuff in this file change exactly?

Reply:
I upgraded `detekt`from `1.6.0` to `1.12.0`, which added and changed a bunch of rules. It seemed simplest to just wipe away the current `detekt` config and regenerate a new default, since I don't think any changes were made to the old default.

Reply:
OK...is that technically a different change from the `buildSrc` updates? I'm just curious if generating the new file could be its own commit to make clear that it's not actually directly related to the core change of the PR. Unless it is.

Reply:
I split this PR into 2 commits:
ba35234 strictly changes the structure, keeping all version numbers the same
4906f26 updates the dependencies, including the update to the `detekt` config.